src/reader.rs

```diff
@@ -543,6 +543,26 @@ pub struct ReaderData {
     in_flight_autosuggest_request: WString,
 
     rls: Option<ReadlineLoopState>,
+    redraw_attempt: RedrawAttempt,
+}
+
+pub struct RedrawAttempt {
+    threshold: usize,
```

Comment:
if we stick with this approach: this field is unused and should be removed (which unlocks further simplification)